src/table.js

```diff
@@ -666,3 +667,79 @@ export function __table(source, operations) {
   }
   return source;
 }
+
+function initKey() {
+  return {
+    other: 0,
+    boolean: 0,
+    integer: 0,
+    number: 0,
+    date: 0,
+    string: 0,
+    array: 0,
+    object: 0,
+    bigint: 0, // TODO for csv, tsv?
+    buffer: 0
+  };
+}
+
+function inferSchema(source) {
+  const schema = [];
+  const sampleSize = 100;
+  const sample = source.slice(0, sampleSize);
+  const typeCounts = {};
+  sample.map((d) => {
+    for (const key in d) {
+      if (!typeCounts[key]) typeCounts[key] = initKey();
+      // for json and sqlite, we already have some types, but for csv and tsv, all
+      // columns are strings here.
+      const type = typeof d[key];
+      const value = type === "string" ? d[key]?.trim() : d[key];
+      if (value === null || value === undefined || value.length === 0)
+        typeCounts[key]["other"]++;
+      else if (type !== "string") {
+        if (Array.isArray(value)) typeCounts[key]["array"]++;
+        else if (value instanceof Date) typeCounts[key]["date"]++;
+        else if (value instanceof ArrayBuffer) typeCounts[key]["buffer"]++;
+        else if (type in typeCounts[key]) typeCounts[key][type]++; // number, bigint, boolean, or object
+      } else {
+        if (value === "true" || value === "false")
+          typeCounts[key]["boolean"]++;
+        else if (!isNaN(+value) && /^-?[0-9]+$/.test(value))
+          typeCounts[key]["integer"]++;
+        else if (!isNaN(+value)) typeCounts[key]["number"]++;
+        else if (
+          value.match(
+            /^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/
+          )
+        )
+          typeCounts[key]["date"]++;
+        else if (value.match(/(\d{1,2})\/(\d{1,2})\/(\d{2,4}) (\d{2}):(\d{2})/))
+          typeCounts[key]["date"]++;
+        else if (value.match(/(\d{4})-(\d{1,2})-(\d{1,2})/))
+          typeCounts[key]["date"]++;
```

Comment:
Copied the lonnng regex from d3's autoType, and added a few more common date formats, but maybe we don't want to be that flexible?

Reply:
Are the types you added appropriately created as dates using `new Date()`? If so, I think that's fine (but maybe arbitrary....?).

Reply:
Yes, they do get created as the expected dates with `new Date()`.

Reply:
In that case, I'd say keep it!